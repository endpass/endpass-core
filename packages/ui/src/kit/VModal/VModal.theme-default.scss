.v-modal-fade-enter-active,
.v-modal-fade-leave-active {
  transition: opacity .5s;
}
.v-modal-fade-enter,
.v-modal-fade-leave-to {
  opacity: 0;
}
