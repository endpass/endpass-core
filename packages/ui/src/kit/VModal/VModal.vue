<template>
  <transition :name="`v-modal-${animation}`">
    <mask-atom @click="$emit('click', $event)">
      <slot />
    </mask-atom>
  </transition>
</template>

<script>
import MaskAtom from '@/atom/mask-atom/mask-atom';

export default {
  name: 'VModal',
  props: {
    animation: {
      type: String,
      default: 'fade',
    },
  },
  components: { MaskAtom },
};
</script>
