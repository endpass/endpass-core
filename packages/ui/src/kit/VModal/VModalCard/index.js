export { default } from './VModalCard.vue';
