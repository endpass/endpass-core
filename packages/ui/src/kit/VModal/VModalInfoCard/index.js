export { default } from './VModalInfoCard.vue';
