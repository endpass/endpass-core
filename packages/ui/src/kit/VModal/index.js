export { default } from './VModal.vue';
