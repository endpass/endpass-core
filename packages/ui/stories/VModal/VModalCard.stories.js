import { storiesOf } from '@storybook/vue';
import { action } from '@storybook/addon-actions';
import VModal from '@/kit/VModal';
import VModalCard from '@/kit/VModal/VModalCard';

const methods = {
  onReturn: action('onReturn'),
  onClose: action('onClose'),
  onMaskClick: action('onMaskClick'),
};

storiesOf('VModal/VModalCard', module)
  .add('default', () => ({
    methods,
    components: { VModal, VModalCard },
    data() {
      return {
        title: 'Modal title',
        body: 'Modal body',
      };
    },
    template: `
      <theme-provider>
        <v-modal @click="onMaskClick">
          <v-modal-card>
            <template slot="title">{{ title }}</template>
            {{ body }}
          </v-modal-card>
        </v-modal>
      </theme-provider>
    `,
  }))
  .add('closable', () => ({
    methods,
    components: { VModal, VModalCard },
    data() {
      return {
        title: 'Modal title',
        body: 'Modal body',
      };
    },
    template: `
      <theme-provider>
        <v-modal @click="onMaskClick">
          <v-modal-card
            :is-closable="true"
            @close="onClose"
          >
            <template slot="title">{{ title }}</template>
            {{ body }}
          </v-modal-card>
        </v-modal>
      </theme-provider>
    `,
  }))
  .add('returnable', () => ({
    methods,
    components: { VModal, VModalCard },
    data() {
      return {
        title: 'Modal title',
        body: 'Modal body',
      };
    },
    template: `
      <theme-provider>
        <v-modal @click="onMaskClick">
          <v-modal-card
            :is-returnable="true"
            @return="onReturn"
          >
            <template slot="title">{{ title }}</template>
            {{ body }}
          </v-modal-card>
        </v-modal>
      </theme-provider>
    `,
  }))
  .add('both', () => ({
    methods,
    components: { VModal, VModalCard },
    data() {
      return {
        title: 'Modal title',
        body: 'Modal body',
      };
    },
    template: `
      <theme-provider>
        <v-modal @click="onMaskClick">
          <v-modal-card
            :is-closable="true"
            @close="onClose"
            :is-returnable="true"
            @return="onReturn"
          >
            <template slot="title">{{ title }}</template>
            {{ body }}
          </v-modal-card>
        </v-modal>
      </theme-provider>
    `,
  }));
