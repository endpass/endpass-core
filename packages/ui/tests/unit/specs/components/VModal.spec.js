import { shallowMount } from '@vue/test-utils';

import VModal from '@/components/VModal';

describe('VModal page', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VModal);
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('VModal');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
