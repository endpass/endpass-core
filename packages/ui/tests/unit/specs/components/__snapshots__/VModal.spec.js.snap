// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VModal page render should render initial state of the component 1`] = `
<div
  class="modal is-active is-footerless"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-logo"
  />
   
  <div
    class="modal-card"
  >
    <div
      class="modal-card-head"
    >
      <h3
        class="modal-card-title"
      />
       
      <button
        class="delete is-large"
      />
    </div>
     
    <div
      class="modal-card-body"
    />
     
    <!---->
  </div>
</div>
`;
